lib/cmdlib/cluster/__init__.py

```diff
@@ -1367,6 +1367,12 @@ def CheckPrereq(self):
             else:
               self.new_os_hvp[os_name][hv_name].update(hv_dict)
 
+      # Cleanup any OS that has an empty hypervisor parameter list, as we don't
+      # need them in the cluster config anymore.
+      for os_name, hvs in self.new_os_hvp.items():
+        if not hvs:
+          self.new_os_hvp.pop(os_name, None)
+
```

Comment:
I have looked further into this. There is currently no accepted way to "remove" an OS image/definition from within ganeti. We keep OS images at /srv/ganeti/os/ and those can be installed/removed from different workflows (deb packages, custom admin scripts, manually, etc). Then we also have os hypervisor parameters which are recorded in the cluster config data. This patch is taking care of cleaning up the latter (not the former) under the assumption that if you have no hypervisors or hypervisor-specific data for a specific OS image, then the OS image should not be present in the config data in the first place (and we have encountered problems already on dangling OS images from config data from images that we had removed previously)